rdkit/Chem/MolStandardize/normalize.py

```diff
@@ -137,33 +137,35 @@ def normalize(self, mol):
         :return: The normalized fragment.
         :rtype: :rdkit:`Mol <Chem.rdchem.Mol-class.html>`
         """
+
         log.debug('Running Normalizer')
         # Normalize each fragment separately to get around quirky RunReactants behaviour
-        fragments = []
-        for fragment in Chem.GetMolFrags(mol, asMols=True):
-            fragments.append(self._normalize_fragment(fragment))
-        # Join normalized fragments into a single molecule again
-        outmol = fragments.pop()
-        for fragment in fragments:
-            outmol = Chem.CombineMols(outmol, fragment)
+        NORMALIZE_FRAGMENT = self._normalize_fragment
+
+        Fragments = Chem.GetMolFrags(mol, asMols=True)
+        outmol = NORMALIZE_FRAGMENT(Fragments[-1])
+        for i in range(len(Fragments) - 2, -1, -1):
+            outmol = Chem.CombineMols(outmol, NORMALIZE_FRAGMENT(Fragments[i]))
```

Comment:
I find this a lot less readable than the original version. 

Considering the rest of the code base, `Fragments` looks like a class or a module name and not a variable name.